x-pack/plugins/ml/public/application/jobs/jobs_list/components/datafeed_modal/datafeed_modal.tsx

```diff
@@ -0,0 +1,281 @@
+  const [selectedTabId, setSelectedTabId] = useState<TabIdsType>(TAB_IDS.CHART);
+  const [isLoadingChartData, setIsLoadingChartData] = useState<boolean>(false);
+  const [bucketData, setBucketData] = useState<number[][]>([]);
+  const [sourceData, setSourceData] = useState<number[][]>([]);
+
+  const { displayErrorToast } = useToastNotificationService();
+  const { euiTheme } = useCurrentEuiTheme();
+  const { job_id: jobId } = datafeedConfig;
+
+  const onSelectedTabChanged = (id: TabIdsType) => {
+    setSelectedTabId(id);
+  };
+
+  const renderTabs = useCallback(
+    () =>
+      tabs.map((tab, index) => (
+        <EuiTab
+          onClick={() => onSelectedTabChanged(tab.id)}
+          isSelected={tab.id === selectedTabId}
+          disabled={tab.disabled}
+          key={index}
+        >
+          {tab.name}
+        </EuiTab>
+      )),
+    [selectedTabId]
+  );
+
+  const handleChange = (date: moment.Moment) => setEndDate(date);
+
+  const handleEndDateChange = (direction: ChartDirectionType) => {
+    const newEndDate = endDate.clone();
+    const [count, type] = interval.split(' ');
+
+    if (direction === CHART_DIRECTION.FORWARD) {
+      newEndDate.add(Number(count), type);
+    } else {
+      newEndDate.subtract(Number(count), type);
+    }
+    setEndDate(newEndDate);
+  };
+
+  const getChartData = useCallback(async () => {
+    const endTimestamp = moment(endDate).valueOf();
+    const [count, type] = interval.split(' ');
+    const startMoment = endDate.clone().subtract(Number(count), type);
+    const startTimestamp = moment(startMoment).valueOf();
+
+    try {
+      const chartData = await ml.results.getDatafeedResultChartData(
+        jobId,
+        timefield,
+        bucketSpan,
+        startTimestamp,
+        endTimestamp,
+        datafeedConfig
+      );
+
+      setSourceData(chartData.datafeedResults);
+      setBucketData(chartData.bucketResults);
+    } catch (error) {
+      const title = i18n.translate('xpack.ml.jobsList.datafeedModal.errorToastTitle', {
+        defaultMessage: 'Error fetching data',
+      });
+      displayErrorToast(error, title);
+    }
+    setIsLoadingChartData(false);
+  }, [endDate, interval]);
+
+  useEffect(
+    function loadChartData() {
+      setIsLoadingChartData(true);
+      getChartData();
+    },
+    [endDate, interval]
+  );
+
+  return (
+    <EuiModal
+      onClose={onClose.bind(null, false)}
+      className="mlDatafeedModal"
+      style={{ height: '600px', minWidth: '800px' }}
+    >
+      <EuiModalHeader>
+        <EuiFlexGroup alignItems="center" justifyContent="spaceBetween" gutterSize="xl">
+          <EuiFlexItem grow={false}>
+            <FormattedMessage
+              id="xpack.ml.jobsList.datafeedModal.header"
+              defaultMessage="{jobId}"
+              values={{
+                jobId,
+              }}
+            />
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiDatePicker
```

Comment:
I wonder if this date picker could do with a label, such as `end time` or `showing data to`. Without a label it needs an aria label I think for accessibility.

![image](https://user-images.githubusercontent.com/7405507/121227634-b0f43f00-c883-11eb-86df-5e629e035d54.png)